qucs/qucs/qucs.cpp

```diff
@@ -834,44 +860,37 @@ void QucsApp::slotSelectComponent(QListWidgetItem *item)
   char *CompFile_cptr;
 
   if (!CompSearch->text().isEmpty()) {
-    //comp search is not empty, search all category to get a component in same name
-    QStringList cats = Category::getCategories();
-    int i = 0;
-    //search normal component
-    foreach (QString it, cats) {
-      Comps = Category::getModules (it);
-      foreach(Module *mod, Comps) {
-        if (mod->info) {
-          (*mod->info)(CompName, CompFile_cptr, false);
-          if (CompName == name) {
-	    // change currently selected category, so the user will 
-	    //   learn where the component comes from
-            CompChoose->setCurrentIndex(i+1); // +1 due to the added "Search Results" item
-	    ccCurIdx = i; // remember the category to select when exiting search
-	    //!! comment out the above two lines if you would like that the search
-	    //!!   returns back to the last selected category instead
-            view->selElem = (*mod->info) (CompName, CompFile_cptr, true);
-	    // TODO: component found, exit the loops now...
-          }
-        }
+    // in "search mode"
+    QVariant v = CompComps->item(i)->data(Qt::UserRole);
+    iconCompInfoStruct iconCompInfo = v.value<iconCompInfoStruct>();
+    qDebug() << "variant :" << v << iconCompInfo.catIdx << iconCompInfo.compIdx;
+
+    Category* cat = Category::Categories.at(iconCompInfo.catIdx);
+    Module *mod = cat->Content.at(iconCompInfo.compIdx);
+    qDebug() << "mod->info" << mod->info;
+    qDebug() << "mod->infoVA" << mod->infoVA;
+    Infos = mod->info;
```

Comment:
(note to self: never again start line-comment threads...)

no need to modify anything. you just need to add the function you need, will not break any of the existing ones.

probably you will not even have to touch module.cpp. the base class implementation should be trivial (no-op), to you can safely put it in the header.